### PR TITLE
Meeting notes

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by favreau** [comment]
### Colormap vs Control points:
- Defining a color map with control points only makes sense if the application shader is capable of computing its values at runtime. In other words, the shader has to be able to compute any value that resolves the polynomial/linear/gaussian/etc function defined by control points. That (heavy and incompatible with realtime rendering) computation has to be implemented for all technologies used by our shaders (c++, glsl, cuda, ispc, etc). If generalization is wanted in the form of a method of the color map event (implemented in C++), that method has to compute the transformation from the control points to a lookup table in order to feed the shaders. And in that case, why not simply expose the color map as a simple lookup table, and leave the control point management to the client application? 
- Why lookup table? Generated by the UI. Used as is by the rendering engines, avoiding any kind of transformation in various technologies. Precision is defined by the sampling. Suitable for realtime rendering.
- Why control points? Precision is infinite if the application can handle them, otherwise a single C++ implementation in the event does the job but looses the interest of control points vs lookup table. More convenient for the UI. Color map is still required for convenience with existing client libraries.

**Comment by favreau** [comment]
After a number of discussions, the conclusion is the following:
- Control points are an abstract representation of the transfer function that do need to be transmitted to the renderer. 
- The main reason  being that renderers will anyway implement a lookup table for performance reasons.
- What we are loosing is the fact that our current UI will not be able to retrieve control points from the renderer itself. But at the same time, the UI is the one that holds the state of the transfer function, and should overwrite anyway any values held by the renderer.
- Sending control points is not enough to define the transfer function. Other information such as tangents and type of algorithm (guassian, linear, etc) also needs to be communicated to the renderer.

To summarize, the transfer function event will be a variable length lookup table (RGBA, and extra attributes) and a range of values to which the lookup table should be applied. The size of the lookup table determines the precision of the sampling.

**Comment by favreau** [comment]
Follow up: https://github.com/HBPVIS/Lexis/pull/20
